library/ssl_tls.c

```diff
@@ -4544,6 +4545,13 @@ int mbedtls_ssl_write_record( mbedtls_ssl_context *ssl, uint8_t force_flush )
 #endif /* MBEDTLS_SSL_DTLS_CONNECTION_ID */
             ssl->out_msglen = len = rec.data_len;
             (void)mbedtls_platform_put_uint16_be( ssl->out_len, rec.data_len );
+            encrypted_fi = 1;
+        }
+
+        //Double check to ensure the encryption has been done
+        if( ssl->transform_out != NULL && encrypted_fi == 0 )
+        {
+            return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
```

Comment:
I think here we could even return `MBEDTLS_ERR_PLATFORM_FAULT_DETECTED` as think can only happen in case of a hardware fault.